modules/post/windows/gather/enum_shares.rb

```diff
@@ -12,174 +12,199 @@ def initialize(info = {})
+      share_info = info.split('\\0')
+      stype = remark = path = nil
+      share_info.each do |e|
+        name, val = e.split('=')
+        case name
+        when 'Path'
+          path = val
+          print_status "\tPath: #{path}"
+        when 'Type'
+          stype = share_type(val.to_i)
+          print_status "\tType: #{stype}"
+        when 'Remark'
+          remark = val
+          print_status("\tRemark: #{remark}") unless remark.blank?
         end
-        # Match the format used by auxiliary/scanner/smb/smb_enumshares
-        # with an added field for path
-        shares << [ sname, stype, remark, path ]
-        print_status()
       end
-      report_note(
-        :host => session,
-        :type => 'smb.shares',
-        :data => { :shares => shares },
-        :update => :unique_data
-      )
-    else
-      print_status("No shares were found")
+
+      print_status
+
+      # Match the format used by auxiliary/scanner/smb/smb_enumshares
+      # with an added field for path
+      shares << [ sname, stype, remark, path ]
     end
+
+    report_note(
+      host: session,
+      type: 'smb.shares',
+      data: { shares: shares },
+      update: :unique_data
+    )
   end
 
   def run
-    print_status("Running against session #{datastore["SESSION"]}")
+    unless datastore['CURRENT'] || datastore['RECENT'] || datastore['ENTERED']
+      fail_with(Failure::BadConfig, 'At least one option (CURRENT, RECENT, ENTERED) must be enabled. Nothing to do.')
+    end
+
+    hostname = sysinfo.nil? ? cmd_exec('hostname') : sysinfo['Computer']
+    print_status("Running module against #{hostname} (#{session.session_host})")
+
+    enum_conf_shares if datastore['CURRENT']
+
+    return unless datastore['RECENT'] || datastore['ENTERED']
 
-    # Variables to hold info
     mount_history = []
     run_history = []
 
-    # Enumerate shares being offered
-    enum_conf_shares() if datastore["CURRENT"]
-    if is_system?
-      mount_history = enum_recent_mounts("HKEY_CURRENT_USER")
-      run_history = enum_run_unc("HKEY_CURRENT_USER")
+    if is_system? || is_admin?
+      mount_history = enum_recent_mounts('HKEY_CURRENT_USER') if datastore['RECENT']
+      run_history = enum_run_unc('HKEY_CURRENT_USER') if datastore['ENTERED']
     else
-      user_sid = []
-      key = "HKU\\"
-      root_key, base_key = session.sys.registry.splitkey(key)
-      open_key = session.sys.registry.open_key(root_key, base_key)
-      keys = open_key.enum_key
-      keys.each do |k|
-        user_sid << k if k =~ /S-1-5-21-\d*-\d*-\d*-\d{3,6}$/
-      end
-      user_sid.each do |us|
-        mount_history = mount_history + enum_recent_mounts("HKU\\#{us.chomp}") if datastore["RECENT"]
-        run_history = run_history + enum_run_unc("HKU\\#{us.chomp}") if datastore["ENTERED"]
+      keys = registry_enumkeys('HKU') || []
+      keys.each do |maybe_sid|
+        next unless maybe_sid.starts_with?('S-1-5-21-')
```

Comment:
Would you mind defining the SID prefix here as a constant? We'll probably end up moving it somewhere eventually where we'll add additional well-known SIDs but for now just marking this one what it is.

`SID_PREFIX_USER`

Reply:
This seems like unnecessary and irrelevant scope creep for this PR but ok.

```# grep -rn S-1-5-2 modules/ lib/
# grep -rn S-1-5-2 modules/ lib/
modules/post/windows/gather/enum_shares.rb:186:        next unless maybe_sid.starts_with?('S-1-5-21-')
modules/post/windows/gather/credentials/nimbuzz.rb:41:      next unless k.include? "S-1-5-21"
modules/post/windows/gather/credentials/idm.rb:45:      next unless k.include? "S-1-5-21"
modules/post/windows/gather/enum_logged_on_users.rb:87:      next unless maybe_sid.starts_with?('S-1-5-21-')
modules/post/windows/manage/ie_proxypac.rb:118:      next unless k.include? "S-1-5-21"
modules/post/windows/manage/ie_proxypac.rb:151:      next unless k.include? "S-1-5-21"
modules/post/windows/manage/ie_proxypac.rb:173:      next unless k.include? "S-1-5-21"
modules/auxiliary/admin/kerberos/ms14_068_kerberos_checksum.rb:46:        OptString.new('USER_SID', [ true, 'The Domain User SID, Ex: S-1-5-21-1755879683-3641577184-3486455962-1000'])
modules/auxiliary/admin/kerberos/ms14_068_kerberos_checksum.rb:54:      print_error("Invalid USER_SID. Ex: S-1-5-21-1755879683-3641577184-3486455962-1000")
lib/msf/core/exploit/remote/kerberos/client/pac.rb:36:            # @option opts [String] :domain_id the domain SID Ex: S-1-5-21-1755879683-3641577184-3486455962
lib/msf/core/post/windows/user_profiles.rb:116:        next unless profkey.starts_with?('S-1-5-21')
```